driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -168,116 +165,75 @@ public void run() {
-      final List<Connection> connections,
-      final List<ListenableFuture<Void>> connectionFutures,
-      final SettableFuture<Void> initFuture,
-      final Queue<Connection> toClose) {
-
-    final Executor initExecutor =
-        manager.cluster.manager.configuration.getPoolingOptions().getInitializationExecutor();
-    final ListenableFuture<List<Void>> allConnectionsFuture = Futures.allAsList(connectionFutures);
-
-    GuavaCompatibility.INSTANCE.addCallback(
-        allConnectionsFuture,
-        new FutureCallback<List<Void>>() {
-          @Override
-          public void onSuccess(List<Void> l) {
-            int added = 0;
-            for (final Connection c : connections) {
-              if (!c.isClosed()) {
-                if (HostConnectionPool.this.connections[c.shardId()].size()
-                    < HostConnectionPool.this.connectionsPerShard) {
-                  ++added;
-                  HostConnectionPool.this.connections[c.shardId()].add(c);
-                } else {
-                  toClose.add(c);
-                }
-              }
-            }
-
-            open.addAndGet(added);
-
-            if (isClosed()) {
-              initFuture.setException(
-                  new ConnectionException(
-                      host.getSocketAddress(), "Pool was closed during initialization"));
-              // we're not sure if closeAsync() saw the connections, so ensure they get closed
-              forceClose(connections);
-              forceClose(toClose);
-              for (List<Connection> shardConnections : HostConnectionPool.this.connections) {
-                forceClose(shardConnections);
-              }
-              open.set(0);
-            } else {
-              int needed = 0;
-              for (final List<Connection> shardsConnections : HostConnectionPool.this.connections) {
-                needed +=
-                    Math.max(
-                        0, HostConnectionPool.this.connectionsPerShard - shardsConnections.size());
-              }
-              if (needed > 0) {
-                int factor = (hostDistance == HostDistance.LOCAL) ? 2 : 1;
-                int limit =
-                    HostConnectionPool.this.connections.length * connectionsPerShard * factor;
-                final List<ListenableFuture<Void>> connectionFutures =
-                    Lists.newArrayListWithCapacity(needed + Math.max(0, toClose.size() - limit));
-                while (toClose.size() > limit) {
-                  connectionFutures.add(toClose.poll().closeAsync());
-                }
-                final List<Connection> connections =
-                    manager.connectionFactory().newConnections(HostConnectionPool.this, needed);
-                for (Connection connection : connections) {
-                  ListenableFuture<Void> connectionFuture = connection.initAsync();
-                  connectionFutures.add(handleErrors(connectionFuture, initExecutor));
+    final Connection connection,
```

Comment:
+1 for sticking with old style of whitespace for the sake of review
once we change style, we shall have it in separate PR again for the sake of review